go/adbc/driver/databricks/database.go

```
@@ -0,0 +1,406 @@
+	}
+
+	// TLS/SSL handling
+	if d.sslMode != "" || d.sslRootCert != "" {
+		var tlsConfig *tls.Config
+
+		// Handle custom root certificate
+		if d.sslRootCert != "" {
+			caCert, err := os.ReadFile(d.sslRootCert)
+			if err != nil {
+				return nil, adbc.Error{
+					Code: adbc.StatusInvalidArgument,
+					Msg:  fmt.Sprintf("failed to read SSL root certificate: %v", err),
+				}
+			}
+
+			caCertPool := x509.NewCertPool()
+			if !caCertPool.AppendCertsFromPEM(caCert) {
+				return nil, adbc.Error{
+					Code: adbc.StatusInvalidArgument,
+					Msg:  "failed to parse SSL root certificate",
+				}
+			}
+
+			tlsConfig = &tls.Config{
+				RootCAs:    caCertPool,
+				MinVersion: tls.VersionTLS12,
+			}
+		}
+
+		// Handle SSL mode
+		if d.sslMode != "" {
+			switch strings.ToLower(d.sslMode) {
+			case "insecure":
+				if tlsConfig == nil {
+					tlsConfig = &tls.Config{MinVersion: tls.VersionTLS12}
+				}
+				tlsConfig.InsecureSkipVerify = true
+			case "require":
+				// Default behavior - full TLS verification
+				if tlsConfig == nil {
+					tlsConfig = &tls.Config{MinVersion: tls.VersionTLS12}
+				}
+			default:
+				return nil, adbc.Error{
+					Code: adbc.StatusInvalidArgument,
+					Msg:  fmt.Sprintf("invalid SSL mode: %s (supported: 'require', 'insecure')", d.sslMode),
+				}
+			}
+		}
+
+		// Apply custom TLS config if we have one
+		if tlsConfig != nil {
+			transport := &http.Transport{
+				TLSClientConfig: tlsConfig,
+			}
+			opts = append(opts, dbsql.WithTransport(transport))
+		}
+	}
+
+	return opts, nil
+}
+
+func (d *databaseImpl) initializeConnectionPool(ctx context.Context) (*sql.DB, error) {
+	opts, err := d.resolveConnectionOptions()
+
+	if err != nil {
+		return nil, err
+	}
+
+	connector, err := dbsql.NewConnector(opts...)
+
+	if err != nil {
+		return nil, err
+	}
+
+	db := sql.OpenDB(connector)
+
+	// Test the connection
+	if err := db.PingContext(ctx); err != nil {
+		_ = db.Close() // Ignore error on cleanup
```

instead of ignoring this error, shouldn't we propagate it?